yb_build_thirdparty_main.py

```diff
@@ -958,6 +958,87 @@ def get_openssl_related_cmake_args(self):
         ]
         return openssl_options
 
+class LibTest:
+    """
+    Verify correct library paths are used in installed dynamically-linked eecutables and
+    libraries.
+    """
+    def __init__(self):
+        self.tp_dir = os.path.dirname(os.path.realpath(sys.argv[0]))
+        self.tp_installed_dir = os.path.join(self.tp_dir, 'installed')
+        if is_mac():
+            self.tool = "otool -L"
+            self.okay_paths = "^\t("
+            self.okay_paths += "/usr/|"
+            self.okay_paths += "/System/Library/|"
+            self.okay_paths += "@rpath|"
+            self.okay_paths += "@loader_path|"
+            self.okay_paths += f"{self.tp_dir})"
+        else:
+            self.tool = "ldd"
+            self.okay_paths = "^\tlinux-vdso|"
+            self.okay_paths += "^\t/lib64/|"
+            self.okay_paths += "^\tstatically linked|"
+            self.okay_paths += "^.* => /lib64/|"
+            self.okay_paths += "^.* => /opt/yb-build/brew/linuxbrew|"
+            self.okay_paths += f"^.* => {self.tp_dir}"
+
+    def good_libs(self,file_path):
+        status = True
+        if is_mac():
+            libout = subprocess.check_output(['otool', '-L' ] + [file_path]).decode('utf-8')
+            if 'is not an object file' in libout:
+                return status
+        else:
+            # SSL libs need to search in common lib
+            ld_search = {"LD_LIBRARY_PATH": self.tp_installed_dir + "/common/lib" }
```

Comment:
We should not have to set LD_LIBRARY_PATH, we should be configuring rpaths correctly. How did this come up?